include/rmm/device_uvector.hpp

```diff
@@ -524,13 +525,21 @@ class device_uvector {
   [[nodiscard]] bool is_empty() const noexcept { return size() == 0; }
 
   /**
-   * @briefreturn{Pointer to underlying resource used to allocate and deallocate the device storage}
+   * @briefreturn{The async_resource_ref used to allocate and deallocate the device storage}
    */
-  [[nodiscard]] mr::device_memory_resource* memory_resource() const noexcept
+  [[nodiscard]] cuda::mr::async_resource_ref<cuda::mr::device_accessible> memory_resource()
```

Comment:
question: Do we need to worry about this change potentially breaking code that was expecting this function to return a `device_memory_resource*`?

I guess the more important question is are we okay with this PR breaking current RMM users or is it intended to be mostly invisible? Its inevitable that we'll have to break people eventually, but I had thought for this initial PR that we were going to try and make it an unobtrusive change.

Reply:
Agree we don't want to break anyone if we can avoid it. Is there a way to convert between these?

Or should we just add a new function `get_resource_ref()` or something like that?

Reply:
We decided that for now we will pull that change out and do the break in a controlled manner